### DISCUSSION TITLE
Row Pinning

### DISCUSSION THREAD

**Discussion by SimaAmini** [TanStack/table#4653] [Q&A]
Is there any built-in feature to pin some of the rows at the top of the table?

**Comment by KevinVandy** [comment]
Not yet, but it is a feature we're considering adding eventually

**Reply by SimaAmini** [reply]
Thanks! I'm looking forward to this new feature.

**Reply by KevinVandy** [reply]
This feature is in v8.10.1

**Reply by zahidiqbalnbs** [reply]
@KevinVandy rows pinning is not working when sorting applied. Can you share some information on that one as well. Or some link to help me how to do that?

**Reply by KevinVandy** [reply]
You'd have to describe your problem in more detail.

**Reply by carlbleick** [reply]
@KevinVandy I would assume that the problem is, that pinned columns are not affected by sorting. I ran into the same issue today and someone else did so too: https://github.com/TanStack/table/discussions/5619.
I could not find any guidance reading the docs and assume that applying the sorting to pinned columns is currently not supported?

**Reply by KevinVandy** [reply]
Ok. Pinned rows are designed to only show up in the order that they were pinned. You could set the pinning state manually with the pinned rows in a custom sorted order.

**Reply by carlbleick** [reply]
Yes that would be a possible workaround, where I would have to apply the current sorting manually to the pinned rows.
